patches/022-openssl-negative-tests.patch

```diff
@@ -0,0 +1,1003 @@
+diff --git a/src/crypto/ecdsa/ecdsa_test.go b/src/crypto/ecdsa/ecdsa_test.go
+index 06957f8c14..7803ae997f 100644
+--- a/src/crypto/ecdsa/ecdsa_test.go
++++ b/src/crypto/ecdsa/ecdsa_test.go
+@@ -9,8 +9,8 @@ import (
+ 	"bytes"
+ 	"compress/bzip2"
+ 	"crypto/elliptic"
+-	"crypto/internal/backend/boringtest"
+ 	boring "crypto/internal/backend"
++	"crypto/internal/backend/boringtest"
+ 	"crypto/internal/bigmod"
+ 	"crypto/rand"
+ 	"crypto/sha1"
+@@ -46,9 +46,6 @@ func testAllCurves(t *testing.T, f func(*testing.T, elliptic.Curve)) {
+ 		tests = append(tests, p224)
+ 	}
+ 	for _, test := range tests {
+-		if boring.Enabled() && !boringtest.Supports(t, "Curve"+test.name) {
+-			t.Skip("unsupported test in FIPS mode")
+-		}
+ 		curve := test.curve
+ 		t.Run(test.name, func(t *testing.T) {
+ 			t.Parallel()
+diff --git a/src/crypto/hmac/hmac.go b/src/crypto/hmac/hmac.go
+index 1b99c68577..3e98ba3717 100644
+--- a/src/crypto/hmac/hmac.go
++++ b/src/crypto/hmac/hmac.go
+@@ -132,6 +132,7 @@ func New(h func() hash.Hash, key []byte) hash.Hash {
+ 		if hm != nil {
+ 			return hm
+ 		}
++		panic("unrecognized hmac in FIPS mode")
+ 		// BoringCrypto did not recognize h, so fall through to standard Go code.
+ 	}
+ 	hm := new(hmac)
+diff --git a/src/crypto/hmac/hmac_test.go b/src/crypto/hmac/hmac_test.go
+index 3898f7f12c..e412a29269 100644
+--- a/src/crypto/hmac/hmac_test.go
++++ b/src/crypto/hmac/hmac_test.go
+@@ -13,6 +13,7 @@ import (
+ 	"crypto/sha512"
+ 	"fmt"
+ 	"hash"
++	"strings"
+ 	"testing"
+ )
+ 
+@@ -549,6 +550,16 @@ var hmacTests = []hmacTest{
+ 
+ func TestHMAC(t *testing.T) {
+ 	for i, tt := range hmacTests {
++		if boring.Enabled() && tt.size == sha1.New().Size() {
```

Comment:
Don't we also want to check `!boringtest.Supports(t, "SHA1")` here?

Reply:
Reworked this.